othertasks/concurrencyExample.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+var filter = ".com"
+
+var wg sync.WaitGroup
+
+// FindCorrectString filters inputed url.
+func FindCorrectString(input string, output chan string) {
+	if strings.Contains(input, filter) {
+		output <- input
+	}
+	defer wg.Done()
+}
+
+// Finder returns urls that contain ".com".
+func Finder(input []string) {
+	results := make(chan string)
+	wg.Add(len(input))
+	for i := range input {
+		go FindCorrectString(input[i], results)
+	}
+	go Result(results)
```

Comment:
change orders of this two lines and try to avoid additional goroutine, deadlock message tells you that channel should be closed.